File: graphql/udemy1-graphql17/GraphQL-JWT-authentication/src/main/java/com/accounts/config/SecurityConfig.java
package com.accounts.config;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
import org.springframework.security.web.SecurityFilterChain;

import static org.springframework.security.config.Customizer.withDefaults;


@Configuration
@EnableWebSecurity
public class SecurityConfig {

    @Autowired
    private AuthEntryPointJwt unauthorizedHandler;

    @Bean
    SecurityFilterChain defaultSecurityFilterChain(HttpSecurity http) throws Exception {

        http.csrf(csrf -> csrf.disable());

        http.authorizeHttpRequests(authorizeRequests ->
                authorizeRequests
                        .requestMatchers("/graphql").permitAll()
                        .anyRequest().authenticated());

        http.exceptionHandling(exception -> exception.authenticationEntryPoint(unauthorizedHandler));

        // Enable basic HTTP authentication
        http.httpBasic(withDefaults());

        return http.build();
    }
}

File: graphql/graphqlex/ex1/src/main/kotlin/com/yscorp/ex1/config/SecurityConfig.kt
package com.yscorp.ex1.config

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.config.Customizer
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
import org.springframework.security.config.annotation.web.configurers.AuthorizeHttpRequestsConfigurer.AuthorizationManagerRequestMatcherRegistry
import org.springframework.security.config.annotation.web.configurers.CsrfConfigurer
import org.springframework.security.config.annotation.web.configurers.ExceptionHandlingConfigurer
import org.springframework.security.web.SecurityFilterChain


@Configuration
@EnableWebSecurity
class SecurityConfig {
    @Autowired
    private val unauthorizedHandler: AuthEntryPointJwt? = null

    @Bean
    @Throws(Exception::class)
    fun defaultSecurityFilterChain(http: HttpSecurity): SecurityFilterChain {
        http.csrf { csrf: CsrfConfigurer<HttpSecurity> -> csrf.disable() }

        http.authorizeHttpRequests { authorizeRequests ->
            authorizeRequests
                .requestMatchers("/graphql").permitAll()
                .anyRequest().authenticated()
        }

        http.exceptionHandling { exception: ExceptionHandlingConfigurer<HttpSecurity?> ->
            exception.authenticationEntryPoint(
                unauthorizedHandler
            )
        }

        // Enable basic HTTP authentication
        http.httpBasic(Customizer.withDefaults())

        return http.build()
    }
}
